week2_rohith_v

Xperior28 opened the pull request.

hrushikeshj commented on week_2/pet.rb:
```diff
@@ -1,115 +1,129 @@
+  def self.group_by_animal_type(pets)
+    names = {}
+    pets.each{|x|
+      names[x.animal_type_id] = x.name
```
Their is a mistake in this logic, but you are close to the answer

hrushikeshj commented on week_2/pet.rb:
```diff
@@ -1,115 +1,129 @@
+  # Return the habitat of the pet
+  def habitat
+    HABITATS.values.each{ |x|
+      x.each{ |k|
```
Prefer `do` for multiline block
```suggestion
      x.each do |k|
```

hrushikeshj commented on week_2/pet.rb:
```diff
@@ -1,115 +1,129 @@
+
+  # Return the habitat of the pet
+  def habitat
+    HABITATS.values.each{ |x|
```
```suggestion
    HABITATS.each do |key, x|
```

hrushikeshj commented on week_2/pet.rb:
```diff
@@ -1,115 +1,129 @@
+  # Pet.cost_to_feed([cat, dog, fish, snake], 6) will return 6180.0
+  def self.cost_to_feed(pets, days)
+    sum = 0
+    pets.each{|x|
+      sum += FOOD_COST_PER_KG[x.animal_type_id]*x.food_consumed_per_day*days
+    }
```
You can make use of the [reduce](https://medium.com/@james.a.hughes/using-the-reduce-method-in-ruby-907f3c18ae1f) method here.

hrushikeshj commented on week_2/prime_numbers.rb:
```diff
@@ -1,18 +1,39 @@
+    if n<=1
+        raise ArgumentError
+      end
+    while i<=n
+        flag = 1
+        j = 2
+        while j<i
+            if(i%j==0)
+                flag = 0;
+                break
+            end
+            j+=1
+        end
+        if flag==1
+            arr.push(i)
+        end
+        i+=1
+    end
+    return arr
```
This solution is correct but u can make it easier to understand by creating a new function to check if an number is prime. And them make use of the [`filter`](https://apidock.com/ruby/Array/filter) method

Xperior28 commented:
I have fixed the group by name function and changed the habitat function
I have also made a seperate function to check prime, to make the code easier to read
You mentioned some changes to be made like implementation of reduce() and filter() function, which I dont fully understand how to use yet, and so I did not implement it in my code for the time being.

hrushikeshj commented on week_2/pet.rb:
```diff
@@ -1,115 +1,132 @@
+  # Note - Order is not important
+  def self.group_by_animal_type(pets)
+    names = {}
+    pets.each{|x|
+      names[x.animal_type_id] = [];
+    }
+    pets.each{|x|
+      names[x.animal_type_id].push(x.name)
+    }
```
This can done in a single loop by checking if a key exists in the has.
```ruby
if names[x.animal_type_id]
  # ....
else
  names[x.animal_type_id] = [x.name]
end
```

Xperior28 replied:
ahh yes 👍
